lib/sequencescape_client.rb

```diff
@@ -35,44 +35,43 @@ def self.update_extraction_attributes(instance, attrs, username = 'test')
   end
 
   def self.purpose_by_name(name)
-    client.plate_purpose.all.select { |p| p.name===name }.first
+    client.plate_purpose.all.find { |p| p.name === name }
   end
 
-  def self.create_plate(purpose_name, attrs)
-    attrs = {}
+  def self.create_plate(purpose_name)
     purpose = purpose_by_name(purpose_name) || purpose_by_name('Stock Plate')
-    purpose.plates.create!(attrs)
-  end
-
-  def self.get_study_by_name(name)
-    get_study_searcher_by_name.first(name: name)
-  rescue Sequencescape::Api::ResourceNotFound => exception
-    return nil
+    purpose.plates.create!({})
   end
 
-  def self.get_study_searcher_by_name
-    @@study_searcher ||= client.search.all.select { |s| s.name == Rails.configuration.searcher_study_by_name }.first
-  end
-
-  def self.get_searcher_by_barcode
-    @@searcher ||= client.search.all.select { |s| s.name == Rails.configuration.searcher_name_by_barcode }.first
-  end
-
-  def self.get_remote_asset(barcode)
+  def self.find_by_barcode(barcode)
     find_by(barcode: barcode)
   end
 
-  def self.find_by_uuid(uuid, opts = nil)
+  def self.find_by_uuid(uuid)
     find_by(uuid: uuid)
   end
 
+  def self.labware(conditions)
+    SequencescapeClientV2::Labware
+      .includes('receptacles.aliquots.sample.sample_metadata,receptacles.aliquots.study,purpose')
+      .select(
+        tubes: SELECT_FOR_IMPORT,
+        plates: SELECT_FOR_IMPORT,
+        tube_racks: 'uuid,labware_barcode,purpose',
+        samples: 'sanger_sample_id,uuid,name',
+        sample_metadata: 'supplier_name,sample_common_name',
+        study: 'name,uuid',
+        purpose: 'name'
+      ).where(**conditions)
+  end
+
   # TODO: In most cases we should know what type of record we're looking up.
   def self.find_by(search_conditions)
     [
-      SequencescapeClientV2::Plate.includes('wells.aliquots.sample'),
-      SequencescapeClientV2::Tube.includes('aliquots.sample'),
-      SequencescapeClientV2::Well.includes('aliquots.sample'),
-      SequencescapeClientV2::TubeRack.includes('racked_tubes.tube.aliquots.sample')
+      SequencescapeClientV2::Plate.includes('wells.aliquots.sample.sample_metadata,wells.aliquots.study,purpose'),
+      SequencescapeClientV2::Tube.includes('aliquots.sample.sample_metadata,aliquots.study'),
+      SequencescapeClientV2::Well.includes('aliquots.sample.sample_metadata,aliquots.study'),
+      SequencescapeClientV2::TubeRack.includes('racked_tubes.tube.aliquots.sample.sample_metadata,racked_tubes.tube.aliquots.study,purpose')
```

Comment:
I've left this as is, to avoid needing to extensively test processes. I know nothing about, but this is relatively inefficient, especially if we know what we are trying to find. Rather than attempting to fix this though I think we'd be better off migrating away from it to better replacements.